### PR TITLE
operator fix

### PR DESCRIPTION
**Pull request by zerlpaMMT**
When using operators in the grid options and the fieldData is an array the export file only displayed "Array" in the column.

### CODE REVIEW

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php [hunk 1]
```diff
@@ -2011,7 +2011,7 @@ protected function getCsvData(Request $request, LocaleServiceInterface $localeSe
                         if (!isset($mappedFieldnames[$field])) {
                             $mappedFieldnames[$field] = $this->mapFieldname($field, $helperDefinitions);
                         }
-                        $objectData[$field] = $fieldData;
+                        $objectData[$field] = is_array($fieldData) ? $fieldData[0] : $fieldData;
```
Comment (weisswurstkanone) [inline bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php:2014]: what is in the rest of the array? can you give an example?

Comment (zerlpaMMT) [reply bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php:2014]: for example an transform operator on quantity value field results in an array with one

File: lib/DataObject/GridColumnConfig/Value/DefaultValue.php [hunk 2]
```diff
@@ -123,21 +123,37 @@ private function getDefaultValue($value)
...
         $brickKey = null;
 
+        
         if (substr($this->attribute, 0, 1) == '~') {
             // key value, ignore for now
+            if(in_array('classificationstore', $attributeParts)){
+                $getter = 'get' . ucfirst($attributeParts[2]);
+                $details = explode('-', $attributeParts[3]);
+
+                $groupId = $this->getGroupByKey($details[1], $element);
```
Comment (weisswurstkanone) [inline lib/DataObject/GridColumnConfig/Value/DefaultValue.php:141]: isn't the group id encoded in $this->attribute ?

File: lib/DataObject/GridColumnConfig/Value/DefaultValue.php [hunk 3]
```diff
@@ -123,21 +123,35 @@ private function getDefaultValue($value)
...
+        
         if (substr($this->attribute, 0, 1) == '~') {
             // key value, ignore for now
+            if(in_array('classificationstore', $attributeParts)){
+                $getter = 'get' . ucfirst($attributeParts[2]);
+                $details = explode('-', $attributeParts[3]);
+
+                if ($element->$getter()->getLocalizedKeyValue($details[0], $details[1], $this->context['language'], true, true)) {
+                    $result = $this->getDefaultValue($element->$getter()->getLocalizedKeyValue($details[0], $details[1], $this->context['language'], true, true)->__toString());
+                } elseif(empty($result) && $element->$getter()->getLocalizedKeyValue($details[0], $details[1], null, true, true)) {
```
Comment (weisswurstkanone) [inline lib/DataObject/GridColumnConfig/Value/DefaultValue.php:143]: this looks odd. why do we need this and what is $result in empty($result) ?

File: bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php (hunk 1 above)
Comment (weisswurstkanone) [reply bundles/AdminBundle/Controller/Admin/DataObject/DataObjectHelperController.php:2014]: how is this related to this PR ?

**Comment by brusch** [comment]
@zerlpaMMT ping 🏓

**Comment by weisswurstkanone** [comment]
Hi @zerlpaMMT , please create a new PR based on our direct discussions.

**Pull request closed without merging**